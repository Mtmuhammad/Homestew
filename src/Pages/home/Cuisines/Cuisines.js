import { React, useState, useEffect } from "react";
import client from "../../../http-common";
import SmallPost from "../../../Components/SmallPost/SmallPost";
import{ Link} from "react-router-dom"
const Cuisines = ({ cuisine, iconColor }) => {
  const [recipes, setRecipes] = useState([]);

  useEffect(() => {
    let saved = false
    //retrieve saved recipes from local storage and save to state
    if (!saved) {
    let data = JSON.parse(localStorage.getItem(`"${cuisine}"`));

    if (data) {
      setRecipes(data);
    } else {
      // retrieve recipes from api if not found in local storage
      client
        .get("complexSearch", {
          params: {
            cuisine,
            apiKey: process.env.REACT_APP_API_KEY,
            number: 4,
          },
        })
        .then(function (res) {
          setRecipes(res.data.results);
          localStorage.setItem(`"${cuisine}"`, JSON.stringify(res.data.results));
        })
        .catch(function (error) {
          console.log(error);
        });
    }
  }
    return () => {
      saved = true
    }
  }, [cuisine]); 

  return (
    <div style={{ maxWidth: "1100px", marginTop: "10rem", marginBottom: "10rem" }}
      className="cuisines container">
      <div className="container">
        <h3 style={{ marginBottom: "2rem" }} className="tertiary">
          {cuisine} recipes{" "}
          <i
            style={{ color: `${iconColor}`, fontSize: "3rem" }}
            className="fa-solid fa-utensils"
          ></i>
        </h3>
        <div className="row">
          {recipes.map((recipe) => (
            <SmallPost
              key={recipe.id}
              id={recipe.id}
              title={recipe.title}
              img={recipe.image}
            />
          ))}
        </div>
        <div className="more text-center mt-5">
          <Link to={`recipes/category/${cuisine}`}>More {cuisine} recipes</Link>
        </div>
      </div>
    </div>
  );
};

export default Cuisines;
