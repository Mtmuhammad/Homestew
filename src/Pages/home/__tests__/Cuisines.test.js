import React from "react";
import { render} from "@testing-library/react";
import Cuisines from "../Cuisines/Cuisines";
import {BrowserRouter} from "react-router-dom"



describe("smoke and snapshot tests", () => {
  test("should render without crashing", async () => {
    render(<Cuisines />, {wrapper: BrowserRouter});
  });

  test("should match snapshot", async () => {
    const { asFragment } = render(<Cuisines />, {wrapper: BrowserRouter});
    expect(asFragment()).toMatchSnapshot();
  });
});

