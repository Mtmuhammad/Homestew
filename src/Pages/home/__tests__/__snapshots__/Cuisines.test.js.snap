// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="cuisines container"
    style="max-width: 1100px; margin-top: 10rem; margin-bottom: 10rem;"
  >
    <div
      class="container"
    >
      <h3
        class="tertiary"
        style="margin-bottom: 2rem;"
      >
         recipes 
        <i
          class="fa-solid fa-utensils"
          style="font-size: 3rem;"
        />
      </h3>
      <div
        class="row"
      />
      <div
        class="more text-center mt-5"
      >
        <a
          href="/recipes/category/undefined"
        >
          More  recipes
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
